### DISCUSSION TITLE
Using ChatGPT custom endpoint

### DISCUSSION THREAD

**Discussion by hozzdutch** [Luke100000/minecraft-comes-alive#890] [General]
I have an open AI account and I've generated a new secret key.
I've tried using the command -
/mca chatAI gpt-3.5-turbo "https://api.openai.com/v1/chat/completions" "secret_key" (replacing secret_key with the key from my chatgpt key
But when I then attempt to talk to a villager, I just get the message "Open AI broke, please send latest.log to dev!"
Am i using this wrong? Is it supposed to be used this way?

**Comment by Luke100000** [comment]
Its surprisingly slow currently but it works for me. Double check your API key.
While I forgot to print the raw response, you can see the error code in the log. If its an 401, the key is invalid for example.

**Comment by hozzdutch** [comment]
This is what the server console outputs when attempting to chat with an NPC (mistral-small works fine, it's just when switching the chatgpt this happens)

17.03 00:34:17 [Server] ForkJoinPool.commonPool-worker-20/ERROR [fo.ne.mc.MCA/]: java.io.IOException: Server returned HTTP response code: 429 for URL: https://api.openai.com/v1/chat/completions

**Reply by hozzdutch** [reply]
a quick google tells me that http code 429 is too many requests. "The HTTP 429 Too Many Requests response status code indicates the user has sent too many requests in a given amount of time ("rate limiting")."
How could this be the case?

**Reply by Luke100000** [reply]
Hmm good question, could it be that openai is currently under increased load? Would explain why its so slow for me as well.

Is a credit card connected? Or do you still have free credits available?

**Reply by hozzdutch** [reply]
Ah got it.
That give me the direction I needed. Thought that just subscribing to chatgpt was enough, didn't realise that I had to buy credits for the API as well. I've now topped that up and now it's all working.
